### PR TITLE
Convert to Polymer 3, add events type definitions

### LINKED ISSUE #156: Polymer 3 conversion
We are going to postpone LitElement conversion until `lit-html` 2.0 is out and Vaadin Fusion is updated to depend on it.

This means the following:

1. The Polymer 3 version will be implemented on top of `master` branch which will also contain changes from `4.3`
2. There will be 5.0.0-alpha1 based on the Polymer 3 and using JS + generated TS definitions.

Once we start updating Vaadin to use lit-html 2.0, the current implementation can be restored.

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #156

### CODE REVIEW

File: src/vaadin-split-layout.js [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+ *   <div>Second</div>
+ * </vaadin-split-layout>
+ * ```
+ *
+ * ### Resize Notification
+ *
+ * This element implements `IronResizableBehavior` to notify the nested resizables
+ * when the splitter is dragged. In order to define a resizable and receive that
+ * notification in a nested element, include `IronResizableBehavior` and listen
+ * for the `iron-resize` event.
```
Comment (tomivirkki) [inline src/vaadin-split-layout.js:144]: This should probably be removed (or at least rephrased)

File: src/vaadin-split-layout.js [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+    this.notifyResize();
+
+    if (event.detail.state === 'end') {
+      this.dispatchEvent(new CustomEvent('splitter-dragend'));
+
+      delete this._startSize;
+    }
+  }
+
```
Comment (tomivirkki) [inline src/vaadin-split-layout.js:343]: Should we still have the explicit `notifyResize` here somewhere...? Now it gets removed from the API docs. Even if we dropped Polymer, we might still want to have the API available?

**Review by tomivirkki — CHANGES_REQUESTED** [review]

File: vaadin-split-layout.d.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export * from './src/vaadin-split-layout.js';
+export * from './@types/interfaces';
```
Comment (tomivirkki) [inline vaadin-split-layout.d.ts:2]: This was removed

Comment (web-padawan) [reply vaadin-split-layout.d.ts]: Good catch, thanks. Removed.

File: src/vaadin-split-layout.js (hunk 2 above)
Comment (web-padawan) [reply src/vaadin-split-layout.js:336]: Restored the method to show it in the API docs.

File: src/vaadin-split-layout.js (hunk 1 above)
Comment (web-padawan) [reply src/vaadin-split-layout.js]: Removed. I don't think we want to highlight `IronResizableBehavior`.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**